src/main/java/com/vinylteam/vinyl/service/impl/DefaultPasswordRecoveryService.java

```diff
@@ -23,88 +23,106 @@
     public void sendLink(String email) {
-        checkForNotNull(email, ErrorRecoveryPassword.EMPTY_EMAIL);
+        checkForIsNotEmptyNotNull(email, PasswordRecoveryError.EMPTY_EMAIL);
         User user = userService.findByEmail(email)
-                .orElseThrow(() -> new RecoveryPasswordException(ErrorRecoveryPassword.EMAIL_NOT_FOUND_IN_DB.getMessage()));
-        RecoveryToken recoveryToken = addRecoveryUserToken(user.getId());
+                .orElseThrow(() -> new RuntimeException(PasswordRecoveryError.EMAIL_NOT_FOUND_IN_DB.getMessage()));
+        RecoveryToken recoveryToken = addRecoveryTokenWithUserId(user.getId());
         sendEmailWithLink(email, recoveryToken.getToken().toString());
     }
 
-    UUID stringToUUD(String token) {
+    UUID stringToUUID(String token) {
         try {
             return UUID.fromString(token);
         } catch (IllegalArgumentException e) {
-            throw new RecoveryPasswordException(ErrorRecoveryPassword.TOKEN_NOT_CORRECT_UUID.getMessage());
+            throw new RuntimeException(PasswordRecoveryError.TOKEN_NOT_CORRECT_UUID.getMessage());
         }
     }
 
     void sendEmailWithLink(String email, String recoveryToken) {
         String recoveryLink = RECOVERY_MESSAGE.replace("{applicationLink}", applicationLink) + recoveryToken;
-        String recoveryTopic = "Recovery Password - WaxDeals";
-        mailSender.sendMail(email, recoveryTopic, recoveryLink);
+        String recoveryTopic = "Password Recovery - WaxDeals";
+        try {
+            mailSender.sendMail(email, recoveryTopic, recoveryLink);
+        } catch (java.lang.RuntimeException e) {
+            throw new RuntimeException(PasswordRecoveryError.EMAIL_SEND_ERROR.getMessage());
+        }
     }
 
     void checkPassword(String newPassword, String confirmPassword) {
-        checkForNotNull(newPassword, ErrorRecoveryPassword.EMPTY_PASSWORD);
+        checkForIsNotEmptyNotNull(newPassword, PasswordRecoveryError.EMPTY_PASSWORD);
         if (!newPassword.equals(confirmPassword)) {
-            throw new RecoveryPasswordException(ErrorRecoveryPassword.PASSWORDS_NOT_EQUAL.getMessage());
+            throw new RuntimeException(PasswordRecoveryError.PASSWORDS_NOT_EQUAL.getMessage());
         }
     }
 
-    void checkForNotNull(String value, ErrorRecoveryPassword emptyValue) {
+    void checkForIsNotEmptyNotNull(String value, PasswordRecoveryError emptyValue) {
         if (value == null || value.isEmpty()) {
-            throw new RecoveryPasswordException(emptyValue.getMessage());
+            throw new RuntimeException(emptyValue.getMessage());
         }
     }
 
-    RecoveryToken addRecoveryUserToken(long userId) {
+    RecoveryToken addRecoveryTokenWithUserId(long userId) {
         UUID token = UUID.randomUUID();
         RecoveryToken recoveryToken = new RecoveryToken();
         recoveryToken.setUserId(userId);
         recoveryToken.setToken(token);
-        passwordRecoveryDao.add(recoveryToken);
+        try {
```

Comment:
Catch blocks have same exception messages. Could you please suggest what is the point to have different exception classes here?